File: ensotrade.mjs
// Minimal client for the EnsoTrade live crypto order-flow API (https://www.ensotrade.tech/v1)
const BASE = "https://www.ensotrade.tech/v1";

const get = async (path) =>
  (await fetch(BASE + path, { headers: { "User-Agent": "ensotrade-api-example" } })).json();

export const explain = (coin) => get(`/explain/${coin}`);   // why a coin is up or down
export const funding = (coin) => get(`/funding/${coin}`);   // funding, crowding, OI, basis
export const orderflow = (coin) => get(`/orderflow/${coin}`); // regime, taker imbalance, OI change

// Demo (run: node ensotrade.mjs)
if (import.meta.url === `file://${process.argv[1]}`) {
  const coin = "btc";
  console.log("WHY:       ", (await explain(coin)).summary);
  console.log("FUNDING:   ", (await funding(coin)).summary);
  console.log("ORDER FLOW:", (await orderflow(coin)).summary);
}

File: ensotrade.py
#!/usr/bin/env python3
"""Minimal client for the EnsoTrade live crypto order-flow API (https://www.ensotrade.tech/v1)."""
import json
import urllib.request

BASE = "https://www.ensotrade.tech/v1"


def get(path: str):
    req = urllib.request.Request(BASE + path, headers={"User-Agent": "ensotrade-api-example"})
    with urllib.request.urlopen(req, timeout=15) as r:
        return json.loads(r.read().decode())


def explain(coin: str):
    """Why a coin is up or down right now, with a quotable summary."""
    return get(f"/explain/{coin}")


def funding(coin: str):
    """Funding rate, crowding, open interest, perp basis."""
    return get(f"/funding/{coin}")


def orderflow(coin: str):
    """Order-flow regime, taker imbalance, open-interest change."""
    return get(f"/orderflow/{coin}")


if __name__ == "__main__":
    coin = "btc"
    print("WHY:       ", explain(coin).get("summary") or explain(coin).get("answer"))
    print("FUNDING:   ", funding(coin).get("summary"))
    print("ORDER FLOW:", orderflow(coin).get("summary"))
